fix: Solve issue with jumpy scrolling on annotation page

teddy-f-47 opened the pull request:
This PR aims to solve the issue with jumpy scrolling behavior on annotation page. The entity editor component has been removed, and now the document text is displayed as a simple paragraph inside the sticky header. Additionally,
 1. A toggle is added to hide/show the progress bar (hiding the progress bar can reduce the area taken by the sticky header).
 2. The navigation buttons in toolbar mobile are now disabled accordingly. Previously, in mobile view, user can still navigate with prev and next buttons, even though they can't confirm any annotation because the confirm button is not available in mobile view. Now the navigation behavior is similar to the desktop view.

Screenshots:
<img width="960" alt="prA" src="https://user-images.githubusercontent.com/64476430/217239832-5e3806ee-b0f8-4d75-bb79-ea1e094cb3a0.PNG">
<img width="960" alt="prB" src="https://user-images.githubusercontent.com/64476430/217239836-5d8ae898-92b7-48f7-934e-94c422f255ac.PNG">

What's changed:
 - frontend/pages/projects/_id/affective-annotation/index.vue : Remove entity editor, remove the onscroll listener for dynamically toggling sticky and hiding the entity editor (now the header is always sticky), add toggle for hiding/showing progress bar
 - frontend/components/tasks/toolbar/ToolbarMobile.vue : Add props to conditionally disable prev and next buttons in mobile view
 - frontend/i18n/* : Add translations for the progress bar toggle label

artikandri commented on frontend/pages/projects/_id/affective-annotation/index.vue:
```diff
@@ -574,16 +528,9 @@ export default {
+      this.showProgressBar = !this.showProgressBar
+    },
 
     getDimensionComponentRef() {
```
Since you are working at it, can you remove this function as well?

teddy-f-47 replied:
I deeply apologize, I didn't realize the fix for the mobile toolbar was included in the refactor PR. 😓 I just found the bug while testing the change in different screen sizes and I thought to just include the fix here since the fix is rather small....

You mean the `getDimensionComponentRef` function, right? Ok i will remove it here

artikandri reviewed (approved):
Hello, thank you for the good work. I like that you have implemented my suggestion to add the toggle button to hide/view the summary in our separate discussion, also to come up with the solution to entirely remove the entity-editor, which is a good solution since we ended up not using the editor and from my multiple tries and experiments, fixing the jumping view without removing it will be harder. 

And as an additional note, I have actually fixed the buttons in the mobile view in my previous PR, but I will remove that adjustment later, should that PR be reviewed. Thanks again and approved.

haanh764 reviewed (approved):
Hello, thank you for your good work. I've seen the code and tested the UI and everything works as intended. I also like the toggle for hiding the progress panel. Approved!

teddy-f-47 commented:
Hi all, I have rebased the branch so that the PR can be merged without conflict :) In addition, I also fixed a minor bug that happened after the rebase, where the document text had a transparent background. Now the document text should be displayed just fine in both light mode and dark mode :)

XeonBloomfield commented:
@teddy-f-47 superb work! Thank you very much for that!
